### DISCUSSION TITLE
Feature Request: Welcome screen

### DISCUSSION THREAD

**Discussion by jeffbcross** [nrwl/nx-console#1024] [Feature Requests]
When I open Nx Console, I just see a list of commands on the left. Users may not know what Nx Console is (like if it was installed from an Extension Pack), or might be unfamiliar with how Nx works. It'd be helpful if there were a landing page or some other component to give the user some context and guidance when they open Nx Console.

**Comment by sandikbarr** [comment]
Honestly, I think rather than needing more guidance, it's the context where we show the Command view with Generate, Run, Build, etc at the top by default, and that isn't the most commonly used feature in a workspace unless you are adding/changing stuff with generators. I would guess the Run/Build/Serve forms are rarely used. And the Common Nx commands view is nice to have, but also not the most common way of running commands either. I think for starters, Projects should be at the top of the views by defaults. This would surface the ability to run targets against your projects, which for me, is the most common thing I point people to first. Especially now that we'll be able to filter that view, it's gonna be so much more useful on large workspaces.

**Comment by sandikbarr** [comment]
It does seem like we are following the extension guidelines for Welcome Views by only showing a welcome view when necessary, which is the 'Select workspace' when your workspace.json is not found at the VS Code workspace root. https://code.visualstudio.com/api/references/extension-guidelines#welcome-views

**Comment by sandikbarr** [comment]
Looking into providing a Welcome Page and Getting Started guide via the walkthroughs contribution point in #1069

**Comment by sandikbarr** [comment]
They are also looking at introducing "What's New" features for showing recent updates to extensions https://github.com/microsoft/vscode/issues/102139

**Reply by sandikbarr** [reply]
The Getting Started experience is now in the Stable release, and they have an example for it as well: https://github.com/microsoft/vscode-extension-samples/tree/main/getting-started-sample
The gist is you add the getting started docs using text and images in the package.json manifest.

**Comment by sandikbarr** [comment]
The Welcome View and Getting Started walkthrough was released in v17.8.0. 

The Getting Started walkthrough should open for users when they first install the extension (which is currently only working for me in Insiders at this time.) The walkthrough is also available from the VS Code Welcome View that opens by default when you open a new VS Code window without a workspace open.
![Screen Shot 2021-08-09 at 10 13 21 AM](https://user-images.githubusercontent.com/2250413/128730485-b0a7440d-6430-46e8-92e0-089a893dcde6.png)

The Getting Started walkthrough is a set of markdown files that contain the same documentation about Nx Console from nx.dev:
![Screen Shot 2021-08-09 at 10 13 38 AM](https://user-images.githubusercontent.com/2250413/128730778-a023eef2-568b-45a5-ae70-5853741b855b.png)

The Welcome View is shown in the Nx Console sidebar when an Angular or Nx workspace is not detected. The Getting Started button is a link to the Nx Console documentation on nx.dev:
![Screen Shot 2021-08-09 at 10 14 46 AM](https://user-images.githubusercontent.com/2250413/128730885-f03e445c-6eae-4e4d-9c10-aee0aa6b7237.png)
